src/main/java/net/earthcomputer/clientcommands/command/CTickCommand.java

```diff
@@ -0,0 +1,210 @@
+import net.minecraft.util.TimeUtil;
+
+import java.text.DecimalFormat;
+
+import static net.fabricmc.fabric.api.client.command.v2.ClientCommandManager.*;
+
+public class CTickCommand {
+
+    private static final DecimalFormat DEC_FMT = new DecimalFormat("0.00");
+
+    private static final String TASK_NAME = "ctick";
+
+    private static TickMeasuringTask currentMeasurer = null;
+
+    public static void register(CommandDispatcher<FabricClientCommandSource> dispatcher) {
+        dispatcher.register(literal("ctick")
+            .then(literal("client")
+                .then(literal("tps")
+                    .executes(ctx -> getClientTps(ctx.getSource())))
+                .then(literal("mspt")
+                    .executes(ctx -> getClientMspt(ctx.getSource()))))
+            .then(literal("server")
+                .then(literal("tps")
+                    .executes(ctx -> getServerTps(ctx.getSource())))
+                .then(literal("mspt")
+                    .executes(ctx -> getServerMspt(ctx.getSource())))));
+    }
+
+    private static int getClientTps(FabricClientCommandSource source) throws CommandSyntaxException {
+        stopPreviousTask();
+
+        TickMeasuringTask measurer = new TickMeasuringTask(true, false);
+        TaskManager.addTask(TASK_NAME, measurer);
+        currentMeasurer = measurer;
+
+        float tps = source.getWorld().tickRateManager().tickrate();
+        source.sendFeedback(Component.translatable("commands.ctick.client.tps.expectedTps", tps));
+        return (int) tps;
+    }
+
+    private static int getClientMspt(FabricClientCommandSource source) throws CommandSyntaxException {
+        stopPreviousTask();
+
+        TickMeasuringTask measurer = new TickMeasuringTask(false, false);
+        TaskManager.addTask(TASK_NAME, measurer);
+        currentMeasurer = measurer;
+
+        float mspt = TimeUtil.MILLISECONDS_PER_SECOND / source.getWorld().tickRateManager().tickrate();
+        source.sendFeedback(Component.translatable("commands.ctick.client.mspt.expectedMspt", mspt));
+        return (int) mspt;
+    }
+
+    private static int getServerTps(FabricClientCommandSource source) throws CommandSyntaxException {
+        stopPreviousTask();
+
+        boolean isIntegratedServer = source.getClient().hasSingleplayerServer();
+        TickMeasuringTask measurer = new TickMeasuringTask(true, !isIntegratedServer);
+        TaskManager.addTask(TASK_NAME, measurer);
+        currentMeasurer = measurer;
+
+        float tps = source.getWorld().tickRateManager().tickrate();
+        source.sendFeedback(Component.translatable("commands.ctick.server.tps.expectedTps", tps));
+        return (int) tps;
+    }
+
+    private static int getServerMspt(FabricClientCommandSource source) throws CommandSyntaxException {
+        stopPreviousTask();
+
+        boolean isIntegratedServer = source.getClient().hasSingleplayerServer();
+        TickMeasuringTask measurer = new TickMeasuringTask(false, !isIntegratedServer);
```

Comment:
I'm not seeing where there is any accurate measurements from the integrated server. Be careful about thread safety with that, though.